rust/agents/relayer/src/msg/pending_message.rs

```diff
@@ -312,9 +310,28 @@ impl PendingOperation for PendingMessage {
                 critical: self.record_message_process_success(),
                 "recording message process success"
             );
+            info!(
+                submission=?self.submission_outcome,
+                "Message successfully processed"
+            );
+            self.submitted = true;
+            self.reset_attempts();
             PendingOperationResult::Success
         } else {
-            self.reset_attempts();
+            if let Some(outcome) = &self.submission_outcome {
+                if let Err(e) = self
+                    .ctx
+                    .origin_gas_payment_enforcer
+                    .record_tx_outcome(&self.message, outcome.clone())
```

Comment:
does this mean that atm we're considering the entire batch's cost as an expenditure for a single message in that batch